source/extensions/filters/http/cors/cors_filter.cc

```diff
@@ -200,11 +205,37 @@ bool CorsFilter::allowCredentials() {
   return false;
 }
 
+std::string CorsFilter::runtimeKey(const std::string& key) {
+  for (const auto policy : policies_) {
+    if (!policy) {
+      continue;
+    }
+    if (!policy->runtimeKey().empty()) {
+      return fmt::format("cors.{}.{}", policy->runtimeKey(), key);
+    }
+  }
+  return EMPTY_STRING;
+}
+
+bool CorsFilter::shadowEnabled() {
+  const auto shadowMode = runtimeKey("shadow_enabled");
+  if (shadowMode == EMPTY_STRING) {
+    return false;
+  }
+  return config_->runtime().snapshot().featureEnabled(shadowMode, 0);
+}
+
 bool CorsFilter::enabled() {
+  const auto filterEnabled = runtimeKey("filter_enabled");
   for (const auto policy : policies_) {
-    if (policy) {
-      return policy->enabled();
+    if (!policy) {
+      continue;
+    }
+    bool policyEnabled = policy->enabled();
+    if (filterEnabled == EMPTY_STRING) {
+      return policyEnabled;
```

Comment:
We can simplify this by dropping the previous `if` and do the following.

```cpp
if (policy && runtimeKey("filter_enabled")) {
  return policy->enabled();
}
```

Reply:
My bad, didn't notice that. You can keep the variables, but I'd simplify the `if` condition as described above.

Reply:
If I simplify the conditional I'll need to also do a check before the variable assignment since I rely on the policy existing.
```cc
    bool policyEnabled = policy->enabled();
    if (filterEnabled == EMPTY_STRING) {
      return policyEnabled;
    }
```

Reply:
Sure. If you return a `absl:optional` from `runtimeKey`, then this will further look like so:

```cpp
bool CorsFilter::enabled() {
  const auto filterEnabled = runtimeKey("filter_enabled");
  for (const auto policy : policies_) {
    if (policy && filterEnabled) {
      return policy->enabled();
    }
    return config_->runtime().snapshot().featureEnabled(filterEnabled.value(), int(policy->enabled()) * 100);
  }
  return false;
}
```

Reply:
Ok, I can inline the policy enabled call. :)